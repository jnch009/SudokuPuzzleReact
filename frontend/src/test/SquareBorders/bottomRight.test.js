import React from 'react';
import { render } from '@testing-library/react';
import 'jest-styled-components';
import SquareBorderBottomRight from '../../Components/Square/squareBorders/bottomRight';

const borderColor = 'blue';

it('Bottom right border', () => {
  const { container } = render(<SquareBorderBottomRight borderColor={borderColor} modify={true} />);
  expect(container.firstChild).toMatchSnapshot();
  expect(container.firstChild).toHaveStyleRule('border-bottom', `0.2em ${borderColor} solid`);
  expect(container.firstChild).toHaveStyleRule('border-right', `0.2em ${borderColor} solid`);
});

it('Bottom right border not modifiable', () => {
  const { container } = render(<SquareBorderBottomRight borderColor={borderColor} modify={false} />);
  expect(container.firstChild).toMatchSnapshot();
  expect(container.firstChild).toHaveStyleRule('border-bottom', `0.2em ${borderColor} solid`);
  expect(container.firstChild).toHaveStyleRule('border-right', `0.2em ${borderColor} solid`);
  expect(container.firstChild).toHaveStyleRule('color', '#fff');
  expect(container.firstChild).toHaveStyleRule('cursor', 'not-allowed!important');
});
