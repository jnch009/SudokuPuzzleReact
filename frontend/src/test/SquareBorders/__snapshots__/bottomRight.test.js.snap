// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bottom right border 1`] = `
.c0 {
  border-bottom: 0.2em blue solid;
  border-right: 0.2em blue solid;
}

<button
  class="c0"
/>
`;

exports[`Bottom right border not modifiable 1`] = `
.c0 {
  border-bottom: 0.2em blue solid;
  border-right: 0.2em blue solid;
  color: #fff;
  cursor: not-allowed!important;
}

<button
  class="c0"
/>
`;
